### PR TITLE
[#1829] Make call-to-action action button configurable

### PR DESCRIPTION
**Pull request by pi-sigma**
Taiga [#1829](https://taiga.maykinmedia.nl/project/open-inwoner/task/1829)

The call-to-action button is made configurable per case status type by adding relevant fields to `ZaakTypeStatusTypeConfig`

### CODE REVIEW

File: src/open_inwoner/openzaak/tests/test_cases.py [hunk 1]
```diff
@@ -194,6 +194,8 @@ def setUpTestData(cls):
             statustype_url=cls.status_type1["url"],
             status_indicator=StatusIndicators.warning,
             status_indicator_text="U moet documenten toevoegen",
+            call_to_action_url="https://example.com",
+            call_to_action_text="yolo",
```
Comment (jiromaykin) [inline src/open_inwoner/openzaak/tests/test_cases.py:198]: Perhaps use "Lorem ipsum" here instead :-)

File: src/open_inwoner/templates/pages/cases/status_inner.html [hunk 2]
```diff
@@ -33,12 +33,59 @@ <h2 class="h2" id="statuses">{% trans 'Status' %}</h2>
...
+                                <li class="{% if case.end_statustype_data %}status--current{% endif %} status-list__list-item status-list__list-item--{{ status.status_indicator }} {% if not case.end_statustype_data %}item--completed status-list__list-item--info{% endif %}">
+                                    {% icon icon="circle" %}
+                                    <div class="status-list__notification status-list__notification--{{ status.status_indicator }}">
+                                        <h3 class="status-list__notification-h3"><a href="#" class="link link-toggle link--bold">{{ status.label }}</a>{% icon icon="expand_more" icon_position="after" outlined=True %}</h3>
+                                        <div class="status-list__notification-content">
+                                            <p class="p p-text p--compact">{{ status.status_indicator_text }}</p>
+                                            <p class="p p-date p--compact">{{ status.date|date }}</p>
+                                            <p>
+                                            {% if case.internal_upload_enabled or case.external_upload_enabled %}
```
Comment (jiromaykin) [inline src/open_inwoner/templates/pages/cases/status_inner.html:47]: This works correctly for status that are currently active, for cases that are not finished yet,
but the button doesn't show up in the last status for Cases that are finalized ('Afgehandeld').
Example: for ISBN login nr 111111110: http://127.0.0.1:8000/mijn-aanvragen/5a7520bb-789d-401f-bdc5-e3a2e92cfeee/status/
Maybe it helps to say I am using this kind of code. in order to show things specifically for the last finalized status only:
`{% if not case.end_statustype_data %}LAST{% endif %}`

**Review by jiromaykin — CHANGES_REQUESTED** [review]

**Review by jiromaykin — COMMENTED** [review]
When testing this: the CTA button needs to be visible/configurable for the current AND final active status, (example: ISBNnr: 111111110, and case: http://127.0.0.1:8000/mijn-aanvragen/054847a7-1ccb-4cd4-8768-1fff29f08cf6/status/)
So: a case that is 'Afgehandeld' will show it's current status as Afgehandled, which is also completed/final. This very last status is a bit of a special animal,
but maybe it will show up correctly in the final 'afgehandeld' status once you've rebased this PR on my PR #821 ?

Also: maybe it's possible to make the final Admin table cells a bit less wide? 
 
Screenshot 2023-11-06 at 16 36 58

**Comment by pi-sigma** [comment]
You need to specify a url for the button, otherwise it won't show (a button without url wouldn't do anything). If you fill in `call_to_action_url` and `call_to_action_text`, it works:
 
Screenshot from 2023-11-07 12-28-12

**Comment by jiromaykin** [comment]
> You need to specify a url for the button, otherwise it won't show (a button without url wouldn't do anything). If you fill in `call_to_action_url` and `call_to_action_text`

Yes, this works - before I approve you may have to remove all of the frontend template code and the styling, or else you're going to have endless merge-conflicts when rebasing, I'd think.

**Comment by alextreme** [comment]
@pi-sigma feel free to resolve the conflicts and assign me to review

File: src/open_inwoner/openzaak/models.py [hunk 3]
```diff
@@ -383,6 +383,26 @@ class ZaakTypeStatusTypeConfig(models.Model):
             "Whether the user should be notfied if a case is set to this type of status"
         ),
     )
+    show_document_uploads = models.BooleanField(
+        default=True,
+        verbose_name=_("Show document uploads"),
+        help_text=_("Wheter document uploads are to be shown"),
```
Comment (alextreme) [inline src/open_inwoner/openzaak/models.py:389]: *whether

File: src/open_inwoner/openzaak/tests/factories.py [hunk 4]
```diff
@@ -115,6 +115,7 @@ class ZaakTypeStatusTypeConfigFactory(factory.django.DjangoModelFactory):
     zaaktype_config = factory.SubFactory(ZaakTypeConfigFactory)
     statustype_url = factory.Faker("url")
     omschrijving = factory.Faker("pystr", max_chars=80)
+    # call_to_action_url = statustype_url
```
Comment (alextreme) [inline src/open_inwoner/openzaak/tests/factories.py:118]: to remove?

**Review by alextreme — APPROVED** [review]
2 trivial remarks, after that this can be merged in my eyes

**Review by alextreme — APPROVED** [review]

**Pull request merged**